### PR TITLE
refactor!: Change to only support compose V2 to generate and run compose files

### LINKED ISSUE #315: Only support use of Docker Compose CLI command
# 🚀 Refactor

### Relevant Package [**REQUIRED**]

 This feature request is for command to run/generate compose files

### Description [**REQUIRED**]
 Old docker-compose tool has been replaced by the new `docker compose` cli command. 

### Describe the solution you'd like
 Replace using `docker-compose` with `docker compose` for generating and running compose files.

### Describe alternatives you've considered
 Have you considered any alternative solutions or workarounds?

### PR DESCRIPTION
**Pull request by lenny-intel**
BREAKING CHANGE: Old docker-compose tool no longer supported. Must use docker compose CLI command. See README for details.

closes #315

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-compose/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  **TBD**

### CODE REVIEW

**Comment by lenny-intel** [comment]
@cherrycl , this is a test PR for generating the compose files using `docker compose` (aka v2). 
Can you determine why it is failing? THX!

**Comment by cherrycl** [comment]
@lenny-intel There are 2 issues caused the failure. 
1. The volume type for device service needs use `bind`, not `volume`.
image
2. TAF file, get-compose-file.sh, also needs to be update. I will open a PR for this issue.

**Comment by lenny-intel** [comment]
@cherrycl , this PR was just to identify any issues. We still plan to release the Levski compose files using `docker-compose` (V1) and make the switch to `V2` for generating in Minnesota. So any teaks to the TAF `get-compose-file.sh` now will need to work with both or you could just submit an issue for the change and do the PR once Levski is released. 

Thanks for looking into this!

**Comment by cherrycl** [comment]
@lenny-intel The TAF change should works on both. The issue [edgex-taf #717](https://github.com/edgexfoundry/edgex-taf/issues/717) was submitted. Will fix it later.

**Comment by cherrycl** [comment]
[TAF issue #717](https://github.com/edgexfoundry/edgex-taf/issues/717) was fixed.

**Comment by bnevis-i** [comment]
@lenny-intel Has this PR served it purpose?

**Comment by lenny-intel** [comment]
@cherrycl , I rebased this PR and it is still not passing the smoke tests even with [TAF issue #717](https://github.com/edgexfoundry/edgex-taf/issues/717) fixed.

Can you please check the failures?

File: taf/docker-compose-taf-no-secty-arm64.yml
Comment (cherrycl) [inline taf/docker-compose-taf-no-secty-arm64.yml:555]:
```suggestion
    - type: bind
```

**Comment by lenny-intel** [comment]
> TAF scripts will override the `PROFILE_VOLUME_PLACE_HOLDER` to `${WORK_DIR}/TAF/config/device-virtual` and `CONF_DIR_PLACE_HOLDER` to `${CONF_DIR}`, both are absolute path.

Ok, I have add a leading `/` so `docker compose` with set the type as bind. The resolved absolute path will start with `//` which should still work.

**Comment by lenny-intel** [comment]
@cherrycl , looks like we are now down to 6  tests (2 unique) that are failing.

**Comment by cherrycl** [comment]
@lenny-intel The failures because we set parameter to `enabled verbose mode` for mqtt-broker by [get-compose-file.sh](https://github.com/edgexfoundry/edgex-taf/blob/03ddac7a5263b1ef5c76871b10e0b56880d07258/TAF/utils/scripts/docker/get-compose-file.sh#L27) to test QoS tests. If I change this command to match this PR, the current Jenkins job will got failures. Do you plan to merge this PR soon? If so I can updated it now or remove this change on `edgex-taf repo` and add the parameter `-v` to enable verbose mode on `edgex-compose repo`.

**Comment by lenny-intel** [comment]
@cherrycl , no I am planning to wait until January to merge after discussion in first TSC of the year.

**Comment by cherrycl** [comment]
@lenny-intel How do you think if I add the `enable verbose mode` parameter on [mqtt-broker file](https://github.com/edgexfoundry/edgex-compose/blob/9b6b23c3653dc23a7e4958a052184a7f4d802159/compose-builder/add-mqtt-broker.yml#L21) on this repo? This can be make both tests on the current job and this PR pass.

**Comment by lenny-intel** [comment]
> @lenny-intel How do you think if I add the `enable verbose mode` parameter on [mqtt-broker file](https://github.com/edgexfoundry/edgex-compose/blob/9b6b23c3653dc23a7e4958a052184a7f4d802159/compose-builder/add-mqtt-broker.yml#L21) on this repo? This can be make both tests on the current job and this PR pass.

What does that look like?

**Comment by cherrycl** [comment]
Update this line
https://github.com/edgexfoundry/edgex-compose/blob/9b6b23c3653dc23a7e4958a052184a7f4d802159/compose-builder/add-mqtt-broker.yml#L21
To
`command: "/usr/sbin/mosquitto -c /mosquitto-no-auth.conf -v"`

**Comment by lenny-intel** [comment]
> command: "/usr/sbin/mosquitto -c /mosquitto-no-auth.conf -v"

Added to verify, but concerned about adding this permanently due to not wanting verbose logs by default. May need to add an option to turn this on that is only used when generating the TAF compose files.

**Comment by lenny-intel** [comment]
@cherrycl , I add the `mqtt-verbose` option and use it when generating the TAF MQTT compose files. Looks like there are other TAF failures now, so can't determine if this verbose option fixes previous failures.

**Comment by cherrycl** [comment]
@lenny-intel TAF script needs to remove the enabling verbose mode command if it is added on edgex-compose. I opened a TAF [PR #752](https://github.com/edgexfoundry/edgex-taf/pull/752) for it. Could you open a standalone PR for `mqtt-verbose`? Otherwise, all QoS related tests for integration-test will fail before this PR is merged.

**Comment by bnevis-i** [comment]
Currently blocked from testing by global configuration changes:

```
Tue Jan  3 15:51:13 UTC 2023 Starting /core-data -cp=consul.http://edgex-core-consul:8500 --registry --confdir=/res ...
flag provided but not defined: -confdir
Usage: /core-data [options]
Server Options:
    -cp, --configProvider           Indicates to use Configuration Provider service at specified URL.
                                    URL Format: {type}.{protocol}://{host}:{port} ex: consul.http://localhost:8500
    -o, --overwrite                 Overwrite configuration in provider with local configuration
                                    *** Use with cation *** Use will clobber existing settings in provider,
                                    problematic if those settings were edited by hand intentionally
    -cf, --configFile <name>        Indicates name of the local configuration file. Defaults to configuration.toml
    -p, --profile <name>            Indicate configuration profile other than default
    -cd, --configDir                Specify local configuration directory
    -r, --registry                  Indicates service should use Registry.

Common Options:
	-h, --help                      Show this message
```

**Comment by lenny-intel** [comment]
@cherrycl , now we are back down to 6 tests failing (3 unique). Can you help root cause these? Thanks!

**Comment by cherrycl** [comment]
> @lenny-intel TAF script needs to remove the enabling verbose mode command if it is added on edgex-compose. I opened a TAF [PR #752](https://github.com/edgexfoundry/edgex-taf/pull/752) for it. Could you open a standalone PR for `mqtt-verbose`? Otherwise, all QoS related tests for integration-test will fail before this PR is merged.

@lenny-intel This failure will be fixed by [edgex-taf #752](https://github.com/edgexfoundry/edgex-taf/pull/752).

**Comment by lenny-intel** [comment]
recheck

**Comment by lenny-intel** [comment]
@cherrycl , on failure left.

**Comment by cherrycl** [comment]
recheck

**Comment by cherrycl** [comment]
@lenny-intel smoke-test has passed.

**Review by bnevis-i — DISMISSED** [review]
LGTM.  Seems pretty solid and I had no problems running it.

**Comment by lenny-intel** [comment]
recheck

**Review by bnevis-i — APPROVED** [review]

**Pull request merged**
